SnapshotTests/SecureConversationsWelcomeScreenLayoutTests.swift

```diff
@@ -0,0 +1,123 @@
+    let theme = Theme.mock()
+
+    func test_welcomeView() {
+        let props = Self.makeWelcomeProps(
+            theme: theme.secureConversationsWelcome,
+            uploads: []
+        )
+        let viewController = SecureConversations.WelcomeViewController(
+            viewFactory: .mock(theme: theme, messageRenderer: nil, environment: .mock),
+            props: .welcome(props),
+            environment: .init(gcd: .live, uiScreen: .mock, notificationCenter: .mock)
+        )
+        viewController.view.frame = UIScreen.main.bounds
+
+        assertSnapshot(
+            matching: viewController.view,
+            as: .image,
+            named: self.nameForDevice()
+        )
+    }
+
+    func test_welcomeWithAttachments() {
+        let props = Self.makeWelcomeProps(
+            theme: theme.secureConversationsWelcome,
+            uploads: [
+                uploadedFileProps(),
+                failedFileUploadProps()
+            ]
+        )
+        let viewController = SecureConversations.WelcomeViewController(
+            viewFactory: .mock(theme: theme, messageRenderer: nil, environment: .mock),
+            props: .welcome(props),
+            environment: .init(gcd: .live, uiScreen: .mock, notificationCenter: .mock)
+        )
+        viewController.view.frame = UIScreen.main.bounds
+
+        assertSnapshot(
+            matching: viewController.view,
+            as: .image,
+            named: self.nameForDevice()
+        )
+    }
+
+    func test_welcomeViewController_withValidationError() {
+        let props = Self.makeWelcomeProps(theme: theme.secureConversationsWelcome, warningMessage: "This is warning message")
+        let viewController = SecureConversations.WelcomeViewController(
+            viewFactory: .mock(theme: theme, messageRenderer: nil, environment: .mock),
+            props: .welcome(props),
+            environment: .init(gcd: .live, uiScreen: .mock, notificationCenter: .mock)
+        )
+        viewController.view.frame = UIScreen.main.bounds
+
+        assertSnapshot(
+            matching: viewController.view,
+            as: .image,
+            named: self.nameForDevice()
+        )
+    }
+
+    // MARK: - Helpers
+
+    func uploadedFileProps() -> SecureConversations.FileUploadView.Props {
+        .init(
+            id: "id-a",
+            style: .messageCenter(theme.secureConversationsWelcome.attachmentListStyle.item),
+            state: .uploaded(.init(localFile: .mock())),
+            removeTapped: .nop
+        )
+    }
+
+    func failedFileUploadProps() -> SecureConversations.FileUploadView.Props {
+        .init(
+            id: "id-b",
+            style: .messageCenter(theme.secureConversationsWelcome.attachmentListStyle.item),
+            state: .error(.network),
+            removeTapped: .nop
+        )
+    }
+
+    static func headerProps() -> Header.Props {
+        .mock(
+            title: "Secure Conversations",
```

Comment:
I guess it is irrelevant which title it is, but the title in the widgets for secure conversations is Messaging

Reply:
Same title is used for VoiceOver tests, so I'd leave it as it is for now, to avoid regenerating new references.